File: client/src/models/reviews/ReviewWithBookViewModel.ts
import { z } from 'zod';
import ReviewUserViewModel from './ReviewUserViewModel';

const ReviewWithBookViewModel = z.object({
  id: z.string().uuid(),
  title: z.string().max(200).nullable(),
  description: z.string().nullable(),
  rating: z.number(),
  user: ReviewUserViewModel,
  created: z.string(),
  book: z.object({
    id: z.string().uuid(),
    title: z.string(),
    yearPublished: z.number().int().positive(),
    coverPictureUrl: z.string().max(256).nullable(),
  }),
});

export default ReviewWithBookViewModel;
export type ReviewWithBookViewModelType = z.infer<typeof ReviewWithBookViewModel>;

File: Server/src/Server.Application/ViewModels/ReviewWithBookViewModel.cs
namespace Server.Application.ViewModels;

public class ReviewWithBookViewModel : ReviewViewModel
{
    public BookInReviewViewModel Book { get; set; } = null!;
}

public sealed class BookInReviewViewModel
{
    public Guid Id { get; set; }
    public string Title { get; set; } = null!;
    public int YearPublished { get; set; }
    public string? CoverPictureUrl { get; set; }
}